pkg/auth/rpc_auth.go

```diff
@@ -112,7 +113,18 @@ func (m *RPCServer) Configure(args ConfigureArg, resp *ConfigureReply) error {
 
 // Authenticate RPCServer Authenticate method
 func (m *RPCServer) Authenticate(args AuthenticateArgs, resp *AuthenticateReply) error {
-	ctx := appctx.PutKeyValuesToCtx(args.Ctx)
+	ctx := setContext(args.Ctx.Token, args.Ctx.User)
 	resp.User, resp.Auth, resp.Error = m.Impl.Authenticate(ctx, args.ClientID, args.ClientSecret)
 	return nil
 }
+
+func setContext(ctxToken []string, userCtx []*userpb.User) context.Context {
```

Comment:
Move this to plugin pkg

Reply:
Thought of doing that, but it leads to cyclic dependencies b/w the user and plugin package. (User imports the plugin package for registering and plugin package would import the user package for the `ContextGetUser` and `ContextSetUser` methods). What do you think about moving all the user-context methods into a separate package, maybe `userctx`? This would involve updating all the packages that use usercontext!

Reply:
Let's have the methods to read and set ctx in `pkg/plugin/ctx`

Reply:
That works, but wouldn't lead to some code duplication?

Reply:
How so?

Reply:
Are you suggesting to *move* the methods from user pkg to `pkg/plugin/ctx`? Because we already have methods to Get and Set contexts in `user` pkg and having the same methods in `pkg/plugin/ctx` would lead to duplication!

Reply:
No it'll have two methods.

```
func GetContextStruct(ctx context.Context) *Ctx
```

which calls ContextGetToken and ContextGetUser, and

```
func GetContext(ctxStruct *Ctx) context.Context
```

which calls ContextSetToken and ContextSetUser.

Reply:
Ahh yes! Was a bit confused. Sorry about that :sweat_smile:

Reply:
There needs to be a small change in the methods. Because `rpc_user.go`, which belongs to the **user** package, will import `GetContextStruct` and `SetContext` methods (for encoding and decoding contexts), we can't have these methods call `ContextSetUser` and `ContextGetUser`. (Cyclic dependencies!). That's why I thought of moving the user context methods out in separate package.

Reply:
Unfortunately having methods to set and read ctx in `pkg/plugin/ctx` won't do us good because of `rpc_user.go`'s dependency on it. So if it is feasible I can send a PR to move out user context methods into a separate pkg.

Reply:
We can do that then

Reply:
Is it fine if I send a separate PR for that?

Reply:
Yes please

Reply:
#1982